File: packages/native-helpers/linux-helper-py/linux_helper/handlers/shortcuts.py
"""Shortcut configuration storage."""

import threading

_lock = threading.Lock()
_shortcuts = {
    "pushToTalk": [],
    "toggleRecording": [],
    "pasteLastTranscript": [],
    "newNote": [],
}


def get_shortcuts() -> dict:
    with _lock:
        return _shortcuts.copy()


def handle_set_shortcuts(params: dict) -> dict:
    global _shortcuts
    with _lock:
        _shortcuts = {
            "pushToTalk": params.get("pushToTalk", []),
            "toggleRecording": params.get("toggleRecording", []),
            "pasteLastTranscript": params.get("pasteLastTranscript", []),
            "newNote": params.get("newNote", []),
        }
    return {"success": True}

File: packages/native-helpers/linux-helper-ts/src/handlers/shortcuts.ts
/**
 * Store configured shortcut key arrays.
 * The evdev monitor uses these to determine which key events
 * are shortcut activations.
 */

interface ShortcutConfig {
  pushToTalk: number[];
  toggleRecording: number[];
  pasteLastTranscript: number[];
  newNote: number[];
}

let shortcuts: ShortcutConfig = {
  pushToTalk: [],
  toggleRecording: [],
  pasteLastTranscript: [],
  newNote: [],
};

export function getShortcuts(): ShortcutConfig {
  return shortcuts;
}

export async function handleSetShortcuts(
  params: Record<string, unknown>,
): Promise<{ success: boolean }> {
  shortcuts = {
    pushToTalk: (params.pushToTalk as number[]) ?? [],
    toggleRecording: (params.toggleRecording as number[]) ?? [],
    pasteLastTranscript: (params.pasteLastTranscript as number[]) ?? [],
    newNote: (params.newNote as number[]) ?? [],
  };
  return { success: true };
}
